src/main/java/de/srendi/advancedperipherals/common/addons/appliedenergistics/AppEngApi.java

```diff
@@ -228,4 +229,205 @@ public static boolean isItemCrafting(MEStorage monitor, ICraftingService grid, I
         return false;
     }
 
+    public static long getTotalItemStorage(IGridNode node) {
+        long total = 0;
+
+        Iterator<IGridNode> iterator = node.getGrid().getMachineNodes(DriveBlockEntity.class).iterator();
+
+        if (!iterator.hasNext()) return 0;
+        while (iterator.hasNext()) {
+            DriveBlockEntity entity = (DriveBlockEntity) iterator.next().getService(IStorageProvider.class);
+            if(entity == null) continue;
+
+            InternalInventory inventory = entity.getInternalInventory();
+
+            for(int i = 0; i < inventory.size(); i++) {
+                ItemStack stack = inventory.getStackInSlot(i);
+
+                if(stack.getItem() instanceof BasicStorageCell cell) {
+                    if(cell.getKeyType().toString().equals("ae2:i")) {
+                        total += cell.getBytes(null);
+                    }
+                } else if(stack.getItem() instanceof DISKDrive disk) {
```

Comment:
A minecraft instance without this mod installed would crash as soon I call this method.
I would recommend to check if the stack is empty, and if it is, continue to the next item.
This would prevent that "instanceof" checks that the "air" item in an empty item stacks is a disk drive


Same for the `listCells` method

Reply:
I added the needed checks.